force-app/main/default/classes/nksGetStoUtilityController.cls

```diff
@@ -203,8 +211,27 @@ public without sharing class nksGetStoUtilityController {
                 AND ServiceChannel.DeveloperName = 'Skriv_til_oss'
                 AND IsReadyForRouting = TRUE
                 AND isPushed = FALSE
+                AND WorkItemId IN :caseIds
             ORDER BY RoutingPriority, SecondaryRoutingPriority DESC NULLS LAST, CreatedDate
-            LIMIT 1
+            LIMIT 5
         ];
+
+        Map<Id, PendingServiceRouting> psrByCaseIdMap = new Map<Id, PendingServiceRouting>();
+        for (PendingServiceRouting psr : psrList) {
+            psrByCaseIdMap.put(psr.WorkItemId, psr);
+        }
+
+        List<Case> cases = [SELECT Id, NKS_Incompetent_IDs__c FROM Case WHERE ID IN :psrByCaseIdMap.keySet()];
+        for (Case casy : cases) {
+            String str = '';
+            try {
+                str += casy.NKS_Incompetent_IDs__c;
+                if (!str.contains(UserInfo.getUserID()))
+                    resList.add(psrByCaseIdMap.get(casy.Id));
+            } catch (Exception ex) {
+                throw new AuraHandledException('Det har oppstått en feil ved henting av ny STO.');
+            }
+        }
+        return resList;
     }
```

Comment:
@SRS-Dxtreme Noe sånt eller du mener at den første delen fjernes? Hvis jeg fjerner den delen, vil det bli komplisert å filtrere listen da vil være behov for flere spørringer (at vi henter hver gang 5 psr og sjekker dem)???

Reply:
Hovedpoenget er at den første spørringen må fjernes da den kommer til å få timeout i produksjon så dette vil ikke fungere, derfor er man nødt til å hente fra X-antall fra køen og filtrere direkte. Hadde vi lagret det i local storage kunne man filtrert i spørringen.